CRM/Mailchimp/Utils.php

```diff
@@ -394,10 +394,6 @@ public static function getGroupsToSync($groupIDs = array(), $mc_list_id = null,
       $params[1] = array($mc_list_id, 'String');
     }
 
-    if ($membership_only) {
-      $whereClause .= " AND (mc_grouping_id IS NULL OR mc_grouping_id = '')";
-    }
-
```

Comment:
Pull and push sync methods uses this flag to push / pull from mailchimp. 
```
CRM/Mailchimp/Form/Sync.php:    $groups = CRM_Mailchimp_Utils::getGroupsToSync(array(), null, $membership_only = TRUE);
CRM/Mailchimp/Form/Pull.php:      $groups = CRM_Mailchimp_Utils::getGroupsToSync(array(), null, $membership_only=TRUE);

```

Not sure about objective of these changes.